### PR TITLE
Find relations to remote events within the new remote events

### PR DESCRIPTION
**Pull request by MadLittleMods**
Find relations to remote events within the new remote events. This allows the reactions to appear in the [Matrix public archive](https://github.com/matrix-org/matrix-public-archive) since we provide one big list of events to display.

Split off from https://github.com/vector-im/hydrogen-web/pull/653

### Dev notes

```
yarn impunity --entry-point src/matrix/room/timeline/Timeline.js --force-esm-dirs lib/ src/
```

Can't run a single test: https://github.com/bwindels/impunity/issues/17

### CODE REVIEW

File: src/matrix/room/timeline/Timeline.js [hunk 1]
```diff
@@ -198,8 +198,10 @@ export class Timeline {
         if (!this._localEntries?.hasSubscriptions) {
             return;
         }
-        // find any local relations to this new remote event
-        for (const pee of this._localEntries) {
+        // find any local relations to these new remote events or maybe these
+        // new remote events reference one of the other new remote events we have.
+        const entryList = new ConcatList(entries, this._localEntries);
+        for (const pee of entryList) {
```
Comment (MadLittleMods) [inline src/matrix/room/timeline/Timeline.js:204]: Failure in CI, https://github.com/vector-im/hydrogen-web/runs/8104935984?check_suite_focus=true#step:7:167

```
 * src::matrix::room::timeline::Timeline::remove remote reaction ... failed
no persisted storage, database can be evicted by browser
AssertionError [ERR_ASSERTION]: 0 == -1
    at remove remote reaction (file:///home/runner/work/hydrogen-web/hydrogen-web/src/matrix/room/timeline/Timeline.js:669:20)
    at async runTests (file:///home/runner/work/hydrogen-web/hydrogen-web/node_modules/impunity/src/runner.mjs:44:5)
    at async main (file:///home/runner/work/hydrogen-web/hydrogen-web/node_modules/impunity/src/main.mjs:21:19)
```

File: src/matrix/room/timeline/Timeline.js [hunk 2]
```diff
@@ -664,7 +666,7 @@ export function tests() {
                 relatedEventId: reactionEntry.id
             }}));
             await poll(() => timeline.entries.length >= 3);
-            assert.equal(messageEntry.pendingAnnotations.get("👋").count, -1);
+            assert.equal(messageEntry.pendingAnnotations.get("👋").count, 0);
```
Comment (MadLittleMods) [inline src/matrix/room/timeline/Timeline.js:669]: AFAICT, the important part of this test is that there are no relations on the message which both `0` or `-1` indicates.

The negative `pendingAnnotations` seems a bit dubious to me but that's a separate issue. Do we really want them to ever be negative?

File: src/matrix/room/timeline/Timeline.js (hunk 1 above)
Comment (MadLittleMods) [reply src/matrix/room/timeline/Timeline.js:204]: Fixed by adjust the value, see https://github.com/vector-im/hydrogen-web/pull/860#discussion_r959095048

File: src/matrix/room/timeline/Timeline.js (hunk 2 above)
Comment (bwindels) [reply src/matrix/room/timeline/Timeline.js:669]: Yeah, the [total annotation count](https://github.com/vector-im/hydrogen-web/blob/master/src/domain/session/room/timeline/ReactionsViewModel.js#L116) is calculated by the remote count + the local, pending count. When removing a remote annotation, the remote count stays 1 until we receive the remote echo. During this time, the pending count is -1 so the total count is 0.

**Review by bwindels — CHANGES_REQUESTED** [review]
As discussed on our pairing session on Friday, I think the http mock approach is better, so if that works out as we prototyped, we won't need this PR anymore.

**Comment by bwindels** [comment]
I'll go ahead and close this PR, given we seem to be committed to the http mock approach.

**Pull request closed without merging**